mantidproject/mantid
# Extend guess range outside of data range in Muon analysis

**Description of work.**

This PR adds the ability to plot the fit guess outside of the range of the data in muon analysis.

**To test:**
1. Open muon analysis.
2. load MUSR 62260-3
3. go to fitting
4. add gausosc function
5. check plot guess by default it plots using the same range as the data
6. in the fit options change the `plot guess using` to x at data points. this allows you to chane the number of points to be used in the guess workspace
7. chane the number of points and the plot will update.
8. change the `plot guess using` to custom x range, this allows you to change the start_x and end_x as well as the number of points in the guess
9. change the start_x and end_x and see the plot update.
10. fit the function
11. got to results and create a results table,
12. go to model fitting, and repete the same steps as for the fitting tab. all the plot guess options should work the same.




Fixes #31876 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
